db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,1627 @@
+package gov.nasa.jpl.aerie.database;
+
+import org.junit.jupiter.api.*;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.concurrent.atomic.AtomicReference;
+
+import gov.nasa.jpl.aerie.database.MerlinDatabaseTestHelper.ExternalEvent;
+import gov.nasa.jpl.aerie.database.MerlinDatabaseTestHelper.ExternalSource;
+import org.postgresql.util.PSQLException;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertThrowsExactly;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SuppressWarnings("SqlSourceToSinkFlow")
```

Comment:
This suppression does nothing currently.

Reply:
Sorry to be a pain, but as a consequence of my request to move some of the insert methods into this class this suppression is needed again. If I had remembered sooner I would've deleted the comment.